src/bm_sim/calculations.cpp

```diff
@@ -360,6 +360,22 @@ struct csum16 {
   }
 };
 
+struct xor16 {
+  uint16_t operator()(const char *buf, size_t len) const {
+    uint16_t mask = 0x00ff;
+    uint16_t final_xor_value = 0x0000;
+    /* Main loop - 2 bytes at a time */
+    for (unsigned int byte = 0; byte < len; byte += 2) {
+      uint16_t t1 = static_cast<uint16_t>(buf[byte]) << 8;
+      uint16_t t2 = 0x0000;
+      /* In case there is a 2nd byte */
+      if ((byte + 1) < len) t2 = static_cast<uint16_t>(buf[byte + 1]);
+      final_xor_value = final_xor_value ^ (t1 + (t2 & mask));
+    }
```

Comment:
Even though this is a xor16 implementation, it's technically possible for len to be odd (e.g. 1, with a single-byte buffer), and you should handle this case (please add a test for this case as well).

Reply:
Thanks for the comment.
The case of len being odd is handled. I have added a test for the same in a new commit.